<a href="https://colab.research.google.com/github/Elvina87/A/blob/main/%D0%94%D0%BE%D0%BC%D0%B0%D1%88%D0%BA%D0%B03.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
DataFrame=pd.read_csv("ds_salaries.csv")

In [ ]:
DataFrame

Unnamed: 0  work_year experience_level employment_type  \
0             0       2020               MI              FT   
1             1       2020               SE              FT   
2             2       2020               SE              FT   
3             3       2020               MI              FT   
4             4       2020               SE              FT   
..          ...        ...              ...             ...   
602         602       2022               SE              FT   
603         603       2022               SE              FT   
604         604       2022               SE              FT   
605         605       2022               SE              FT   
606         606       2022               MI              FT   

                      job_title  salary salary_currency  salary_in_usd  \
0                Data Scientist   70000             EUR          79833   
1    Machine Learning Scientist  260000             USD         260000   
2             Big Data Engineer   85000             GBP         109024   
3          Product Data Analyst   20000             USD          20000   
4     Machine Learning Engineer  150000             USD         150000   
..                          ...     ...             ...            ...   
602               Data Engineer  154000             USD         154000   
603               Data Engineer  126000             USD         126000   
604                Data Analyst  129000             USD         129000   
605                Data Analyst  150000             USD         150000   
606                AI Scientist  200000             USD         200000   

    employee_residence  remote_ratio company_location company_size  
0                   DE             0               DE            L  
1                   JP             0               JP            S  
2                   GB            50               GB            M  
3                   HN             0               HN            S  
4                   US            50               US            L  
..                 ...           ...              ...          ...  
602                 US           100               US            M  
603                 US           100               US            M  
604                 US             0               US            M  
605                 US           100               US            M  
606                 IN           100               US            L  

[607 rows x 12 columns]

In [ ]:
DataFrame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Unnamed: 0          607 non-null    int64 
 1   work_year           607 non-null    int64 
 2   experience_level    607 non-null    object
 3   employment_type     607 non-null    object
 4   job_title           607 non-null    object
 5   salary              607 non-null    int64 
 6   salary_currency     607 non-null    object
 7   salary_in_usd       607 non-null    int64 
 8   employee_residence  607 non-null    object
 9   remote_ratio        607 non-null    int64 
 10  company_location    607 non-null    object
 11  company_size        607 non-null    object
dtypes: int64(5), object(7)
memory usage: 57.0+ KB


 С помощью команды .info мы получили информации о нашем датафрейме. В столбце Non-Null указывается количество ненулевых значений. В датафрейме 607 строк (и 12 столбцов), а это значит, что все ячейки таблицы не пустые.

In [ ]:
DataFrame.isnull()

Unnamed: 0  work_year  experience_level  employment_type  job_title  \
0         False      False             False            False      False   
1         False      False             False            False      False   
2         False      False             False            False      False   
3         False      False             False            False      False   
4         False      False             False            False      False   
..          ...        ...               ...              ...        ...   
602       False      False             False            False      False   
603       False      False             False            False      False   
604       False      False             False            False      False   
605       False      False             False            False      False   
606       False      False             False            False      False   

     salary  salary_currency  salary_in_usd  employee_residence  remote_ratio  \
0     False            False          False               False         False   
1     False            False          False               False         False   
2     False            False          False               False         False   
3     False            False          False               False         False   
4     False            False          False               False         False   
..      ...              ...            ...                 ...           ...   
602   False            False          False               False         False   
603   False            False          False               False         False   
604   False            False          False               False         False   
605   False            False          False               False         False   
606   False            False          False               False         False   

     company_location  company_size  
0               False         False  
1               False         False  
2               False         False  
3               False         False  
4               False         False  
..                ...           ...  
602             False         False  
603             False         False  
604             False         False  
605             False         False  
606             False         False  

[607 rows x 12 columns]

Так же для поиска пустых (NaN) ячеек в датафрейме можно использовать методы библиотеки Pandas. Используем метод -  .isnull(). Метод выполняет одно и то же действие и возвращают булевский датафрейм той же формы, что и исходный, но с True в местах, где значения пустые, и False в местах, где значения не пустые.

Попробуем отфильтровать строчки с пустыми ячейками по столбцам
work_year: год трудоустройства,
зарплата_in_usd: Эквивалентная сумма зарплаты в долларах США (долларах США),
Remote_ratio: процент удаленной работы, разрешенный для данной должности.

In [ ]:
DataFrame.isnull().loc[DataFrame['remote_ratio']==True]

Empty DataFrame
Columns: [Unnamed: 0, work_year, experience_level, employment_type, job_title, salary, salary_currency, salary_in_usd, employee_residence, remote_ratio, company_location, company_size]
Index: []

In [ ]:
DataFrame.isnull().loc[DataFrame['salary_in_usd']==True]

Empty DataFrame
Columns: [Unnamed: 0, work_year, experience_level, employment_type, job_title, salary, salary_currency, salary_in_usd, employee_residence, remote_ratio, company_location, company_size]
Index: []

In [ ]:
DataFrame.isnull().loc[DataFrame['work_year']==True]

Empty DataFrame
Columns: [Unnamed: 0, work_year, experience_level, employment_type, job_title, salary, salary_currency, salary_in_usd, employee_residence, remote_ratio, company_location, company_size]
Index: []

Поиск пустых ячеек подтвердил нам, что их в нашем массиве нет.

**Посмотрим на корреляцию:**

In [ ]:
DataFrame.corr()

<ipython-input-27-b50b12afc3ad>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  DataFrame.corr()


Unnamed: 0  work_year    salary  salary_in_usd  remote_ratio
Unnamed: 0       1.000000   0.886550 -0.096250       0.167025      0.095000
work_year        0.886550   1.000000 -0.087577       0.170493      0.076314
salary          -0.096250  -0.087577  1.000000      -0.083906     -0.014608
salary_in_usd    0.167025   0.170493 -0.083906       1.000000      0.132122
remote_ratio     0.095000   0.076314 -0.014608       0.132122      1.000000

"Unnamed: 0" (порядковый номер) и	"work_year" (год) можно не брать в расчет, а именно коэффициент 0,88655 на их пересечении.
Остальные все коэффициенты корреляции имеют маленькие значения, близкие к нулю, то есть можно сказать, что связь отсутствует.

---



**Применим функции сложения, нахождения среднего значения, медианы, минимума и максимума, стандартного отклонения для столбца "зарплата_in_usd" из датасета.**

In [ ]:
DataFrame['salary_in_usd'].sum()

68164807

In [ ]:
DataFrame['salary_in_usd'].mean()

112297.86985172982

In [ ]:
DataFrame['salary_in_usd'].median()

101570.0

In [ ]:
DataFrame['salary_in_usd'].std()

70957.25941139569

In [ ]:
DataFrame['salary_in_usd'].min()

2859

In [ ]:
DataFrame['salary_in_usd'].max()

600000

**Применим операции сравнения.**

In [ ]:
Выведем информацию по 2022г

In [ ]:
DataFrame[DataFrame['work_year'] > 2021]

Unnamed: 0  work_year experience_level employment_type       job_title  \
289         289       2022               SE              FT   Data Engineer   
290         290       2022               SE              FT    Data Analyst   
291         291       2022               SE              FT    Data Analyst   
292         292       2022               MI              FT  Data Scientist   
293         293       2022               MI              FT  Data Scientist   
..          ...        ...              ...             ...             ...   
602         602       2022               SE              FT   Data Engineer   
603         603       2022               SE              FT   Data Engineer   
604         604       2022               SE              FT    Data Analyst   
605         605       2022               SE              FT    Data Analyst   
606         606       2022               MI              FT    AI Scientist   

     salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
289  135000             USD         135000                 US           100   
290  155000             USD         155000                 US           100   
291  120600             USD         120600                 US           100   
292  130000             USD         130000                 US             0   
293   90000             USD          90000                 US             0   
..      ...             ...            ...                ...           ...   
602  154000             USD         154000                 US           100   
603  126000             USD         126000                 US           100   
604  129000             USD         129000                 US             0   
605  150000             USD         150000                 US           100   
606  200000             USD         200000                 IN           100   

    company_location company_size  
289               US            M  
290               US            M  
291               US            M  
292               US            M  
293               US            M  
..               ...          ...  
602               US            M  
603               US            M  
604               US            M  
605               US            M  
606               US            L  

[318 rows x 12 columns]

Выведем информацию, где "зар плата в долларах" больше средней зарплаты

In [ ]:
DataFrame[DataFrame['salary_in_usd'] > DataFrame['salary_in_usd'].mean()]

Unnamed: 0  work_year experience_level employment_type  \
1             1       2020               SE              FT   
4             4       2020               SE              FT   
6             6       2020               SE              FT   
8             8       2020               MI              FT   
9             9       2020               SE              FT   
..          ...        ...              ...             ...   
602         602       2022               SE              FT   
603         603       2022               SE              FT   
604         604       2022               SE              FT   
605         605       2022               SE              FT   
606         606       2022               MI              FT   

                      job_title  salary salary_currency  salary_in_usd  \
1    Machine Learning Scientist  260000             USD         260000   
4     Machine Learning Engineer  150000             USD         150000   
6           Lead Data Scientist  190000             USD         190000   
8         Business Data Analyst  135000             USD         135000   
9            Lead Data Engineer  125000             USD         125000   
..                          ...     ...             ...            ...   
602               Data Engineer  154000             USD         154000   
603               Data Engineer  126000             USD         126000   
604                Data Analyst  129000             USD         129000   
605                Data Analyst  150000             USD         150000   
606                AI Scientist  200000             USD         200000   

    employee_residence  remote_ratio company_location company_size  
1                   JP             0               JP            S  
4                   US            50               US            L  
6                   US           100               US            S  
8                   US           100               US            L  
9                   NZ            50               NZ            S  
..                 ...           ...              ...          ...  
602                 US           100               US            M  
603                 US           100               US            M  
604                 US             0               US            M  
605                 US           100               US            M  
606                 IN           100               US            L  

[270 rows x 12 columns]

Выведем информацию, где "Remote_ratio: процент удаленной работы" от 0 до 75.

In [ ]:
 DataFrame[(DataFrame['remote_ratio']> 0) & (DataFrame['remote_ratio'] < 75)]

Unnamed: 0  work_year experience_level employment_type  \
2             2       2020               SE              FT   
4             4       2020               SE              FT   
7             7       2020               MI              FT   
9             9       2020               SE              FT   
15           15       2020               MI              FT   
..          ...        ...              ...             ...   
487         487       2022               EN              PT   
495         495       2022               MI              FT   
498         498       2022               SE              FT   
505         505       2022               EN              FT   
511         511       2022               MI              FT   

                      job_title    salary salary_currency  salary_in_usd  \
2             Big Data Engineer     85000             GBP         109024   
4     Machine Learning Engineer    150000             USD         150000   
7                Data Scientist  11000000             HUF          35735   
9            Lead Data Engineer    125000             USD         125000   
15                 Data Analyst      8000             USD           8000   
..                          ...       ...             ...            ...   
487              Data Scientist    100000             USD         100000   
495  Machine Learning Scientist    153000             USD         153000   
498          Research Scientist     85000             EUR          93427   
505              Data Scientist    120000             AUD          86703   
511       Business Data Analyst     90000             CAD          70912   

    employee_residence  remote_ratio company_location company_size  
2                   GB            50               GB            M  
4                   US            50               US            L  
7                   HU            50               HU            L  
9                   NZ            50               NZ            S  
15                  PK            50               PK            L  
..                 ...           ...              ...          ...  
487                 DZ            50               DZ            M  
495                 US            50               US            M  
498                 FR            50               FR            L  
505                 AU            50               AU            M  
511                 CA            50               CA            L  

[99 rows x 12 columns]

Выведем информацию, где "Remote_ratio: процент удаленной работы" равен 75.

In [ ]:
DataFrame[DataFrame['remote_ratio'] == 75 ]

Empty DataFrame
Columns: [Unnamed: 0, work_year, experience_level, employment_type, job_title, salary, salary_currency, salary_in_usd, employee_residence, remote_ratio, company_location, company_size]
Index: []